modules/rostests/apitests/version/VerQueryValue.c

```diff
@@ -0,0 +1,302 @@
+                .wValueLength = 0,
+                .wType = 1,
+                .szKey = L"Comments",
+            },
+            .Comments2 =
+            {
+                .wLength = 28,
+                .wValueLength = 0,
+                .wType = 1,
+                .szKey = L"Comments2",
+                .Padding = 0xCC,
+            },
+            .FooBar =
+            {
+                .wLength = 28,
+                .wValueLength = 4,
+                .wType = 1,
+                .szKey = L"FooBar",
+                .Value = L"Bar",
+            },
+        },
+    },
+};
+
+void
+Test_VerQueryValueA(void)
+{
+    DWORD dwVersionInfoSizeW, dwVersionInfoSizeA;
+    PVOID pvVersionInfoA, pvData = NULL;
+    CHAR szSubBlock[256];
+    USHORT uLanguage, uCodePage;
+    UINT cbLen;
+    BOOL result;
+
+    /* Get the size for the unicode version info */
+    dwVersionInfoSizeW = GetFileVersionInfoSizeW(L"kernel32.dll", NULL);
+    ok(dwVersionInfoSizeW > 0, "GetFileVersionInfoSizeW failed\n");
+
+    /* Get the size for the ANSI version info */
+    dwVersionInfoSizeA = GetFileVersionInfoSizeA("kernel32.dll", NULL);
+    ok(dwVersionInfoSizeA > 0, "GetFileVersionInfoSizeA failed\n");
+    ok(dwVersionInfoSizeA == dwVersionInfoSizeW, "Unexpected size\n");
+
+    /* Get the ANSI version info from kernel32 */
+    pvVersionInfoA = malloc(dwVersionInfoSizeA);
+    memset(pvVersionInfoA, 0xCC, dwVersionInfoSizeA);
+    result = GetFileVersionInfoA("kernel32.dll", 0, dwVersionInfoSizeA, pvVersionInfoA);
+    ok(result, "GetFileVersionInfoA failed\n");
+
+    /* Query available translations */
+    result = VerQueryValueA(pvVersionInfoA,
+                            "\\VarFileInfo\\Translation",
+                            &pvData,
+                            &cbLen);
+    ok(result, "VerQueryValueA failed\n");
+    ok(cbLen >= 4, "Unexpected value\n");
+    ok((cbLen & 0x3) == 0, "Unexpected value\n");
+    uLanguage = ((USHORT*)pvData)[0];
+    uCodePage = ((USHORT*)pvData)[1];
+
+    /* Query sublock */
+    sprintf(szSubBlock, "\\StringFileInfo\\%04X%04X\\CompanyName", uLanguage, uCodePage);
+    result = VerQueryValueA(pvVersionInfoA,
+                            szSubBlock,
+                            &pvData,
+                            &cbLen);
+    ok(result, "VerQueryValueA failed\n");
+    ok(cbLen >= 2, "Unexpected value\n");
+    ok((cbLen & 0x1) == 0, "Unexpected value\n");
+
+    free(pvVersionInfoA);
+}
+
+void
+Test_VerQueryValueW(void)
+{
+    DWORD dwVersionInfoSizeW;
+    PVOID pvVersionInfoW, pvData = NULL;
+    WCHAR szSubBlock[256];
+    USHORT uLanguage, uCodePage;
+    UINT cbLen;
+    BOOL result;
+
+    /* Get the size for the unicode version info */
+    dwVersionInfoSizeW = GetFileVersionInfoSizeW(L"kernel32.dll", NULL);
+    ok(dwVersionInfoSizeW > 0, "GetFileVersionInfoSizeW failed\n");
+
+    /* Get the unicode version info from kernel32 */
+    pvVersionInfoW = malloc(dwVersionInfoSizeW);
```

Comment:
The ansi test also add a filling of the allocated buffer with `0xCC`.  Why not doing it there as well?

Reply:
The ansi test calls `GetFileVersionInfoA` to fill the buffer appropriately. The static test passes something that is already supposed to be filled in.